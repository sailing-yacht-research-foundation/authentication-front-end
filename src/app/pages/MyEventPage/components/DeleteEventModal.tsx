import React from 'react';
import { Modal } from 'antd';
import { toast } from 'react-toastify';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import { deleteEvent } from 'services/live-data-server/event-calendars';
import { translations } from 'locales/translations';
import { showToastMessageOnRequestError } from 'utils/helpers';

export const DeleteEventModal = (props) => {

    const { t } = useTranslation();

    const { onRaceDeleted } = props;

    const {
        event,
        showDeleteModal,
        setShowDeleteModal
    } = props;

    const [isLoading, setIsLoading] = React.useState<boolean>(false);

    const performDeleteRace = async () => {
        setIsLoading(true);
        const response = await deleteEvent(event.id);
        setIsLoading(false);

        setShowDeleteModal(false);

        if (response.success) {
            toast.success(t(translations.delete_event_modal.successfully_deleted, { name: event.name }));
            onRaceDeleted();
        } else {
            showToastMessageOnRequestError(response.error);
        }
    }

    return (
        <Modal
            confirmLoading={isLoading}
            title={t(translations.delete_event_modal.are_you_sure_you_want_to_delete)}
            visible={showDeleteModal}
            onOk={() => {
                performDeleteRace();
            }}
            onCancel={() => {
                setShowDeleteModal(false);
            }}>
            <ModalMessage>{t(translations.delete_event_modal.you_will_delete)}</ModalMessage>
        </Modal>
    )
}

const ModalMessage = styled.div`
    margin: 0 5px;
    margin-bottom: 15px;
`;