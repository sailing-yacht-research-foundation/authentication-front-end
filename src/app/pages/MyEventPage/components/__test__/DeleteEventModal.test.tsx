import * as React from 'react';
import { DeleteEventModal } from '../DeleteEventModal';
import * as DeleteEventModalModule from '../DeleteEventModal';
import { createRenderer } from 'react-test-renderer/shallow';
import { i18n } from 'locales/i18n';
import { translations } from 'locales/translations';
import { act, fireEvent, render, } from '@testing-library/react';

const shallowRenderer = createRenderer();

const modalShowComponent = (
    <DeleteEventModal event={{}}
        onRaceDeleted={jest.fn}
        showDeleteModal={false}
        setShowDeleteModal={jest.fn} />
);

describe('DeleteEventModal', () => {
    it('should render and match snapshot', () => {
        shallowRenderer.render(modalShowComponent);
        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });

    // Use 'it' to test a single attribute of a target
    it('Hide if show modal flag is false', () => {

        shallowRenderer.render(modalShowComponent);

        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput.props.visible).not.toBeTruthy();
    });

    it('Show if show modal flag is true', async () => {
        shallowRenderer.render(
            <DeleteEventModal event={{}}
                onRaceDeleted={jest.fn}
                showDeleteModal={true}
                setShowDeleteModal={jest.fn} />);

        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput.props.visible).toBeTruthy();
    });

    it('Modal title and content should match i18next translation language', async () => {
        const t = await i18n;

        shallowRenderer.render(
            <DeleteEventModal event={{}}
                onRaceDeleted={jest.fn}
                showDeleteModal={true}
                setShowDeleteModal={jest.fn} />);

        expect(t(shallowRenderer.getRenderOutput().props.title)).toBe(t(translations.delete_event_modal.are_you_sure_you_want_to_delete));
        expect(t(shallowRenderer.getRenderOutput().props.children.props.children)).toBe(t(translations.delete_event_modal.you_will_delete));
    });

    it("It should have been rendered with correct received props", () => {
        const deleteEventModalSpy = jest.spyOn(DeleteEventModalModule, 'DeleteEventModal');

        render(<DeleteEventModal event={{}}
            onRaceDeleted={jest.fn}
            showDeleteModal={true}
            setShowDeleteModal={jest.fn} />);

        expect(deleteEventModalSpy).toHaveBeenCalledWith({
            event: {},
            onRaceDeleted: jest.fn,
            showDeleteModal: true,
            setShowDeleteModal: jest.fn
        }, {});
    });

    it("It should call setShowDeleteModal when pressing cancel text", () => {
        const setShowDeleteModalMock = jest.fn();

        const { getByText } = render(
            <DeleteEventModal event={{}}
                onRaceDeleted={jest.fn}
                showDeleteModal={true}
                setShowDeleteModal={setShowDeleteModalMock} />
        );

        act(() => {
            fireEvent.click(getByText('Cancel'));
        });

        expect(setShowDeleteModalMock).toHaveBeenCalledWith(false);
    });
});
