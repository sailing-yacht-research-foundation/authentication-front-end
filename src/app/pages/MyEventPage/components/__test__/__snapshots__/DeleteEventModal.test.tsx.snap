// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEventModal should render and match snapshot 1`] = `
<Modal
  confirmLoading={false}
  okType="primary"
  onCancel={[Function]}
  onOk={[Function]}
  title="Are you sure you want to delete this event?"
  visible={false}
  width={520}
>
  <styled.div>
    You will delete this event along with all associated information, are you sure you want to continue?
  </styled.div>
</Modal>
`;
